File: greedy/best-time-to-buy-and-sell-stock-ii.java
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length;

        int minPrice = Integer.MAX_VALUE;
        int totalProfit = 0;


        for (int i = 0; i < n; i++) {
            if (prices[i] <= minPrice) {
                minPrice = prices[i];
            } else {
                totalProfit += prices[i] - minPrice;
                minPrice = prices[i];
            }

        }

        return totalProfit;
        
    }
}

File: greedy/best-time-to-buy-and-sell-stock-ii.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        length = len(prices)
        for i in range(length - 1):
            if prices[i + 1] > prices[i]:
                profit += prices[i + 1] - prices[i]
        
        return profit
                